src/extension.ts

```diff
@@ -100,18 +100,25 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
             await ext.tree.refresh(ext.attachedAccountsNode);
         });
         registerCommand('azureDatabases.refresh', async (_actionContext: IActionContext, node?: AzExtTreeItem) => await ext.tree.refresh(node));
-        registerCommand('cosmosDB.detachDatabaseAccount', async (actionContext: IActionContext, node?: AzureTreeItem) => {
-            if (!node) {
-                node = await ext.tree.showTreeItemPicker<AzureTreeItem>(cosmosDBTopLevelContextValues.map((val: string) => val += AttachedAccountSuffix), actionContext);
-            }
-            if (node instanceof MongoAccountTreeItem) {
-                if (ext.connectedMongoDB && node.fullId === ext.connectedMongoDB.parent.fullId) {
-                    setConnectedNode(undefined, codeLensProvider);
-                    await node.refresh();
+        registerCommand('cosmosDB.detachDatabaseAccount', async (actionContext: IActionContext & ITreeItemPickerContext, node?: AzureTreeItem) => {
+            const children = await ext.attachedAccountsNode.loadAllChildren(actionContext);
+            if (!("connectionString" in children[0])) {
+                const message = localize('noAttachedAccounts', 'There are no Attached Accounts.');
+                vscode.window.showInformationMessage(message);
+                actionContext.suppressCreatePick = true;
```

Comment:
We should be able to remove this line since the tree item picker isn't being shown